app/addons/documents/changes/changes-header.react.jsx

```diff
@@ -0,0 +1,236 @@
+  var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
+
+
+  var ChangesHeader = React.createClass({
+    getInitialState: function () {
+      return {
+        showTabContent: changesFilterStore.isTabVisible()
+      };
+    },
+
+    componentDidMount: function () {
+      changesFilterStore.on('change', this.onChange, this);
+    },
+
+    onChange: function () {
+      this.setState({
+        showTabContent: changesFilterStore.isTabVisible()
+      });
+    },
+
+    toggleFilterSection: function () {
+      Actions.toggleTabVisibility();
+    },
+
+    render: function () {
+      var tabContent = '';
+
+      if (this.state.showTabContent) {
+        tabContent = <ChangesHeaderTabContent key="changesFilterSection" />;
+      }
+
+      return (
+        <div className="changes-header-section">
+          <ChangesHeaderTab onToggle={this.toggleFilterSection} />
+          <ReactCSSTransitionGroup transitionName="toggleChangesFilter" component="div" className="changes-tab-content">
+            {tabContent}
+          </ReactCSSTransitionGroup>
+        </div>
+      );
+    }
+  });
+
+  var ChangesHeaderTab = React.createClass({
+    propTypes: {
+      onToggle: React.PropTypes.func.isRequired
+    },
+
+    render: function () {
+      return (
+        <div className="dashboard-upper-menu">
+          <ul className="nav nav-tabs" id="db-views-tabs-nav">
+            <li>
+              <a href="#filter" onClick={this.props.onToggle} data-bypass="true" data-toggle="tab">
+                <i className="fonticon fonticon-plus"></i> Filter
+              </a>
+            </li>
+          </ul>
+        </div>
+      );
+    }
+  });
+
+  var ChangesHeaderTabContent = React.createClass({
+    render: function () {
+      return (
+        <div className="tab-content">
+          <div className="tab-pane active" ref="filterTab">
+            <div className="changes-header js-filter">
+              <ChangesFilter />
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var ChangesFilter = React.createClass({
+    getStoreState: function () {
+      return {
+        filters: changesFilterStore.getFilters()
+      };
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    // props belong to the parent, so these aren't stored in the related store
+    getDefaultProps: function () {
+      return {
+        addItemPlaceholder: 'Type a filter',
+        helpText: 'e.g. debug or <1.4.1> or any regex',
+        tooltip: ''
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    submitForm: function (e) {
+      e.preventDefault();
+      e.stopPropagation();
+
+
+      var filter = this.refs.addItem.getDOMNode().value.trim();
+      if (!filter) {
+        return;
+      }
+
+      Actions.addFilterViewItem(filter);
+      this.refs.addItem.getDOMNode().value = '';
```

Comment:
@robertkowalski is correct. This could be simplified to how we assigned a value from onChange